src/test/java/com/eleks/academy/whoami/service/GameServiceImplTest.java

```diff
@@ -35,21 +45,33 @@ public void setMockMvc() {
 
 	@Test
 	void createGameTest() {
+		final String idNaming = "id";
 		final String player = "player";
+		final String expectedGameStatus = WaitingForPlayers.class.getName();
+
 		SynchronousGame game = new PersistentGame(player, gameRequest.getMaxPlayers());
 
 		when(gameRepository.save(any(SynchronousGame.class))).thenReturn(game);
 
-		var createdGame = gameService.createGame(player, gameRequest);
-		NewGameRequest gameRequest1 = new NewGameRequest();
-		gameRequest1.setMaxPlayers(4);
+		var gameDetails = gameService.createGame(player, gameRequest);
+
+		var expectedGame = GameDetails.builder()
+				.status(expectedGameStatus)
+				.players(List.of(new PlayerWithState(new PersistentPlayer(player), PlayerState.READY)))
+				.build();
+
+		assertThat(gameDetails)
+				.usingRecursiveComparison()
+				.ignoringFields(idNaming)
+				.isEqualTo(expectedGame);
+
+		assertNotNull(game.getId());
+		assertEquals(game.getId(), gameDetails.getId());
 
-		assertThat(gameService.createGame(player, gameRequest))
-				.isEqualTo(gameService.createGame("player", gameRequest1));
-		assertNotNull(createdGame.getId());
-		assertNotNull(createdGame.getStatus());
+		assertNotNull(game.getStatus());
```

Comment:
better to check with specific value, not just it is null

Reply:
Resolved